labs/basics/basics2/mainwindow.cpp

```diff
@@ -0,0 +1,150 @@
+    setCentralWidget(widget_);
+    this->Action();
+    options_ = menuBar()->addMenu(tr("&Options"));
+    auto* action = new QAction("&Mode 1");
+    auto* action2 = new QAction("&Mode 2");
+    options_->addAction(action);
+    options_->addAction(action2);
+    connect(action, &QAction::triggered, this, &MainWindow::Action);
+    connect(action2, &QAction::triggered, this, &MainWindow::Action2);
+    setWindowTitle(tr("The Omniscient app"));
+    setMinimumSize(160, 160);
+    resize(960, 640);
+}
+
+void MainWindow::Action() {
+    this->clear();
+    widget_ = new QWidget;
+    setCentralWidget(widget_);
+    label_ = new CustomLabel;
+    question_ = new QLineEdit;
+    button_ = new QPushButton("Click to reply");
+    answer_ = new Answer;
+    QVBoxLayout* layout = new QVBoxLayout;
+    QVBoxLayout* layout2 = new QVBoxLayout;
+    label_->setText("Enter your question");
+    label_->setMaximumHeight(40);
+    label_->setAlignment(Qt::AlignCenter);
+    answer_->resize(200,200);
+    answer_->setStyleSheet("background-color: gray;");
+    answer_->setAlignment(Qt::AlignCenter);
+    layout2->addWidget(label_);
+    layout2->addWidget(question_);
+    layout2->addWidget(button_);
+    layout->addLayout(layout2);
+    layout->addWidget(answer_);
+    widget_->setLayout(layout);
+    // options_ = menuBar()->addMenu(tr("&Options"));
+    // auto* action = new QAction("&Mode 1");
+    // auto* action2 = new QAction("Mode 2");
+    // options_->addAction(action);
+    // options_->addAction(action2);
+    connect(button_, &QPushButton::clicked, answer_, &Answer::GenerateAnswer);
+    connect (button_, &QPushButton::clicked, label_, &CustomLabel::ChangeText1);
+    connect (question_, &QLineEdit::editingFinished, label_, &CustomLabel::ChangeText2);
+}
+
+void MainWindow::Action2() {
+    this->clear();
+    widget_ = new QWidget;
+    setCentralWidget(widget_);
+    label2_ = new QLabel("Enter your numbers");
+    label2_->setMaximumHeight(30);
+    line2_ = new QLineEdit;
+    line2_->setMinimumHeight(200);
+    line2_->setAlignment(Qt::AlignLeft);
+    line2_->setAlignment(Qt::AlignTop);
+    combo_box2_ = new QComboBox;
+    combo_box2_->addItem("sequence");
+    combo_box2_->addItem("one number");
+    button2_ = new QPushButton ("Click to randomize");
+    sequence_ = new RandomField(this);
+    sequence_->setMinimumHeight(200);
+    sequence_->setStyleSheet("background-color: gray; color: white");
+    sequence_->setAlignment(Qt::AlignCenter);
+    QVBoxLayout* layout = new QVBoxLayout;
+    layout->addWidget(label2_);
+    layout->addWidget(line2_);
+    layout->addWidget(combo_box2_);
+    layout->addWidget(button2_);
+    layout->addWidget(sequence_);
+    widget_->setLayout(layout);
+    connect (button2_, &QPushButton::clicked, sequence_, &RandomField::Random);
+}
+
+void RandomField::Random() {
+    auto elements = parent_->Sequence()->text().split(" ");
+    std::random_shuffle(elements.begin(), elements.end());
```

Comment:
```suggestion
    std::random_device rd;
    std::mt19937 g(rd());
    std::shuffle(elements.begin(), elements.end(), g);
```

https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2014/n4190.htm